src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -43,11 +44,12 @@ public static IEnumerable<IPackage> GetPackages(ChocolateyConfiguration configur
                 results = results.Where(p => p.IsLatestVersion);
             }
 
-            return results.OrderBy(p => p.Id)
-                          .AsEnumerable()
-                          .Where(PackageExtensions.IsListed)
-                          .Where(p => configuration.Prerelease || p.IsReleaseVersion())
-                          .distinct_last(PackageEqualityComparer.Id, PackageComparer.Version).ToList();
+            // just trust the server order, don't sort because that's a blocking operation
+            // also don't worry about multiple versions, considering Is*LatestVersion only applies to one

```

Comment:
LOL -- yes, that is what ensures only one not all (for file based repos)

It's a blocking filter, which means that when you're calling web API based repositories (where it's not necessary), you're going to have to wait for ALL results to be received and parsed before you can output any results. It's dramatically slower to first output (and thus _feels_ a lot slower), but **critical*** for file folder repositories.